metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -135,14 +136,21 @@ final class RenameProvider(
         val textEdits = for (loc <- locs) yield {
           textEdit(isOccurence, loc, params.getNewName())
         }
-        Seq(uri -> textEdits.toList)
+        Seq(uri.toAbsolutePath -> textEdits.toList)
       }
       val fileChanges = allChanges.flatten.toMap
-      val (openedEdits, closedEdits) = fileChanges.partition {
-        case (file, edits) =>
-          val path = AbsolutePath(Paths.get(new URI(file)))
-          buffers.contains(path)
-      }
+      val (openedEdits, closedEdits) =
+        if (!metalsConfig.openFilesOnRenames || fileChanges.keySet.size >= metalsConfig.renameFileThreshold) {
+          if (metalsConfig.openFilesOnRenames) {
```

Comment:
I find this check a bit confusing: we're checking for `!metalsConfig.openFilesRenames` and immediately after for `metalsConfig.openFilesOnRenames`. Can we simplify this somehow?

Reply:
I extraced `shouldRenameInBackground` value and it should be a bit clearer I think.